server/battleships/utils/src/main/java/eu/optas/utils/Converter.java

```diff
@@ -0,0 +1,16 @@
+package eu.optas.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class Converter<I, O> {
+    public List<O> convertAll(List<I> input) {
+        List<O> output = new ArrayList<>();
```

Comment:
gali iškart basics mokintis javos Stream API.
Šitą vietą galima būtų perdaryti taip:
return input.stream()
                  .map(this::convert)
                  .collect(Collectors.toList());
                  
Čia labai panašiai kaip ir javascripte su collectionais dirbi